src/client/protocol.c

```diff
@@ -14,67 +14,258 @@
 #include "../tracing.h"
 #include "../tuple.h"
 
+/* Convert a value that potentially borrows data from the client_proto read buffer
+ * into one that owns its data. The owned data must be free with freeOwnedValue. */
+static int makeValueOwned(struct value *val)
+{
+	char *p;
+	switch (val->type) {
```

Comment:
I like a `default` case in a switch statement, I also think Coverity will complain if it's absent.